runtimes/talos/src/main/java/com/talosvfx/talos/runtime/graphics/NineSlice.java

```diff
@@ -0,0 +1,363 @@
+
+        if (top > 0) {
+            if (left > 0) patches[TOP_LEFT] = new TextureRegion(region, 0, 0, left, top);
+            if (middleWidth > 0) patches[TOP_CENTER] = new TextureRegion(region, left, 0, middleWidth, top);
+            if (right > 0) patches[TOP_RIGHT] = new TextureRegion(region, left + middleWidth, 0, right, top);
+        }
+        if (middleHeight > 0) {
+            if (left > 0) patches[MIDDLE_LEFT] = new TextureRegion(region, 0, top, left, middleHeight);
+            if (middleWidth > 0) patches[MIDDLE_CENTER] = new TextureRegion(region, left, top, middleWidth, middleHeight);
+            if (right > 0) patches[MIDDLE_RIGHT] = new TextureRegion(region, left + middleWidth, top, right, middleHeight);
+        }
+        if (bottom > 0) {
+            if (left > 0) patches[BOTTOM_LEFT] = new TextureRegion(region, 0, top + middleHeight, left, bottom);
+            if (middleWidth > 0) patches[BOTTOM_CENTER] = new TextureRegion(region, left, top + middleHeight, middleWidth, bottom);
+            if (right > 0) patches[BOTTOM_RIGHT] = new TextureRegion(region, left + middleWidth, top + middleHeight, right, bottom);
+        }
+
+        // If split only vertical, move splits from right to center.
+        if (left == 0 && middleWidth == 0) {
+            patches[TOP_CENTER] = patches[TOP_RIGHT];
+            patches[MIDDLE_CENTER] = patches[MIDDLE_RIGHT];
+            patches[BOTTOM_CENTER] = patches[BOTTOM_RIGHT];
+            patches[TOP_RIGHT] = null;
+            patches[MIDDLE_RIGHT] = null;
+            patches[BOTTOM_RIGHT] = null;
+        }
+        // If split only horizontal, move splits from bottom to center.
+        if (top == 0 && middleHeight == 0) {
+            patches[MIDDLE_LEFT] = patches[BOTTOM_LEFT];
+            patches[MIDDLE_CENTER] = patches[BOTTOM_CENTER];
+            patches[MIDDLE_RIGHT] = patches[BOTTOM_RIGHT];
+            patches[BOTTOM_LEFT] = null;
+            patches[BOTTOM_CENTER] = null;
+            patches[BOTTOM_RIGHT] = null;
+        }
+    }
+
+    /** Construct a degenerate "nine" patch with only a center component. */
+    public NineSlice(Texture texture, Color color) {
+        this(texture);
+        setColor(color);
+    }
+
+    public NineSlice(Texture texture) {
+        this(new TextureRegion(texture));
+    }
+
+    /** Construct a degenerate "nine" patch with only a center component. */
+    public NineSlice(TextureRegion region) {
+        this(//
+                null, null, null, //
+                null, region, null, //
+                null, null, null //
+        );
+    }
+
+    /** Construct a nine patch from the given nine texture regions. The provided patches must be consistently sized (e.g., any left
+     * edge textures must have the same width, etc). Patches may be <code>null</code>. Patch indices are specified via the public
+     * members {@link #TOP_LEFT}, {@link #TOP_CENTER}, etc. */
+    public NineSlice(TextureRegion... patches) {
```

Comment:
No reason for this to be.... Its not varying args

Reply:
fixed